test/unittest/blas3/blas3_gemm_common.hpp

```diff
@@ -383,4 +396,327 @@ static std::string generate_batched_strided_name(
   BLAS_REGISTER_TEST_CUSTOM_NAME(test_suite, test_suite##combination, \
                                  verify_gemm,                         \
                                  gemm_batched_strided_arguments_t,    \
-                                 combination, generate_batched_strided_name);
\ No newline at end of file
+                                 combination, generate_batched_strided_name);
+
+#ifdef BLAS_ENABLE_COMPLEX
+
+template <typename scalar_t, helper::AllocType mem_alloc>
+inline void verify_gemm(const gemm_cplx_arguments_t<scalar_t> arguments) {
+  std::string alloc;
+  index_t offset;
+  index_t batch;
+  index_t m;
+  index_t n;
+  index_t k;
+  char transa;
+  char transb;
+  std::complex<scalar_t> alpha;
+  std::complex<scalar_t> beta;
+  index_t lda_mul;
+  index_t ldb_mul;
+  index_t ldc_mul;
+  gemm_batch_type_t batch_type;
+  std::tie(alloc, offset, batch, m, n, k, transa, transb, alpha, beta, lda_mul,
+           ldb_mul, ldc_mul, batch_type) = arguments;
+
+  const char ta_str[2] = {transa, '\0'};
+  const char tb_str[2] = {transb, '\0'};
+
+  auto q = make_queue();
+  blas::SB_Handle sb_handle(q);
+
+  const index_t lda = ((transa != 'n') ? k : m) * lda_mul;
+  const index_t ldb = ((transb != 'n') ? n : k) * ldb_mul;
+  const index_t ldc = m * ldc_mul;
+
+  const index_t size_a = m * k * lda_mul;
+  const index_t size_b = k * n * ldb_mul;
+  const index_t size_c = m * n * ldc_mul;
+
+  const index_t buffer_size_a = batch * size_a + offset;
+  const index_t buffer_size_b = batch * size_b + offset;
+  const index_t buffer_size_c = batch * size_c + offset;
+
+  std::vector<std::complex<scalar_t>> a_m(buffer_size_a);
+  std::vector<std::complex<scalar_t>> b_m(buffer_size_b);
+  std::vector<std::complex<scalar_t>> c_m_gpu(buffer_size_c);
+
+  fill_random(a_m);
+  fill_random(b_m);
+  fill_random(c_m_gpu);
+  std::vector<std::complex<scalar_t>> c_m_cpu = c_m_gpu;
+
+  // Use system blas to create a reference output
+  for (int i = 0; i < batch; ++i) {
+    reference_blas::cgemm<scalar_t>(
+        ta_str, tb_str, m, n, k, reinterpret_cast<const void*>(&alpha),
+        reinterpret_cast<const void*>(a_m.data() + i * size_a + offset), lda,
+        reinterpret_cast<const void*>(b_m.data() + i * size_b + offset), ldb,
+        reinterpret_cast<const void*>(&beta),
+        reinterpret_cast<void*>(c_m_cpu.data() + i * size_c + offset), ldc);
+  }
+
+  if (batch > 1 && batch_type == gemm_batch_type_t::interleaved) {
```

Comment:
We could move this if condition to before the call to `reference_blas` to avoid running the `cgemm` unnecessarily.

Reply:
Yes done!